opentelemetry-prometheus/src/lib.rs

```diff
@@ -80,447 +94,557 @@
-const SCOPE_INFO_DESCRIPTION: &str = "Instrumentation Scope metadata";
-/// Create a new prometheus exporter builder.
-pub fn exporter(controller: BasicController) -> ExporterBuilder {
-    ExporterBuilder::new(controller)
+/// Creates a builder to configure a [PrometheusExporter]
+pub fn exporter() -> ExporterBuilder {
+    ExporterBuilder::default()
 }
 
-/// Configuration for the prometheus exporter.
+/// Prometheus metrics exporter
 #[derive(Debug)]
-pub struct ExporterBuilder {
-    /// The prometheus registry that will be used to register instruments.
-    ///
-    /// If not set a new empty `Registry` is created.
-    registry: Option<prometheus::Registry>,
-
-    /// The metrics controller
-    controller: BasicController,
-
-    /// config for exporter
-    config: Option<ExporterConfig>,
+pub struct PrometheusExporter {
+    reader: Arc<ManualReader>,
 }
 
-impl ExporterBuilder {
-    /// Create a new exporter builder with a given controller
-    pub fn new(controller: BasicController) -> Self {
-        ExporterBuilder {
-            registry: None,
-            controller,
-            config: Some(Default::default()),
-        }
-    }
-
-    /// Set the prometheus registry to be used by this exporter
-    pub fn with_registry(self, registry: prometheus::Registry) -> Self {
-        ExporterBuilder {
-            registry: Some(registry),
-            ..self
-        }
-    }
-
-    /// Set config to be used by this exporter
-    pub fn with_config(self, config: ExporterConfig) -> Self {
-        ExporterBuilder {
-            config: Some(config),
-            ..self
-        }
-    }
-
-    /// Sets up a complete export pipeline with the recommended setup, using the
-    /// recommended selector and standard processor.
-    pub fn try_init(self) -> Result<PrometheusExporter, MetricsError> {
-        let config = self.config.unwrap_or_default();
-
-        let registry = self.registry.unwrap_or_else(prometheus::Registry::new);
-
-        let controller = Arc::new(Mutex::new(self.controller));
-        let collector =
-            Collector::with_controller(controller.clone()).with_scope_info(config.with_scope_info);
-        registry
-            .register(Box::new(collector))
-            .map_err(|e| MetricsError::Other(e.to_string()))?;
-
-        let exporter = PrometheusExporter {
-            registry,
-            controller,
-        };
-        global::set_meter_provider(exporter.meter_provider()?);
-
-        Ok(exporter)
-    }
-
-    /// Sets up a complete export pipeline with the recommended setup, using the
-    /// recommended selector and standard processor.
-    ///
-    /// # Panics
-    ///
-    /// This panics if the exporter cannot be registered in the prometheus registry.
-    pub fn init(self) -> PrometheusExporter {
-        self.try_init().unwrap()
+impl TemporalitySelector for PrometheusExporter {
+    fn temporality(&self, kind: InstrumentKind) -> Temporality {
+        self.reader.temporality(kind)
```

Comment:
nit: probably worth to add a comment here that prometheus only supports Cumulative temporality so the return value here will always be Cumulative(the default value of ManualReader)